### PR TITLE
Add GHC 9.4 support

### PR DESCRIPTION
**Pull request by parsonsmatt**

### CODE REVIEW

**Review by vrom911 — APPROVED** [review]
Looks good, thanks!

**Comment by locallycompact** [comment]
Can this be merged?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ jobs:
           - "8.10.7"
           - "9.0.2"
           - "9.2.1"
+          - "9.4.1"
```
Comment (jhrcek) [inline .github/workflows/ci.yml:29]: I think at this point we could switch the CI to 9.2.4 and 9.4.2 respectively..

**Comment by jhrcek** [comment]
I'd love to get this merged. This is also blocking co-log and co-log-core from building with ghc 9.4.2. :pray:

**Comment by parsonsmatt** [comment]
Closing in favor of #124 which is much nicer imo.

**Pull request closed without merging**